source/plugin/commands/commandbuilding.rst

```diff
@@ -0,0 +1,200 @@
+
+Player-Only commands
+~~~~~~~~~~~~~~~~~~~~
+
+Sometimes it is necessary that only players can execute a command. Example ``/suicide`` command. 
+This is best done though the ``executionRequirements`` on the command's builder where we check for
+a target player on the command cause. 
+
+.. code-block:: java
+
+    public Command.Parameterized buildCommand(){
+        return Command
+            .builder()
+            .executor(new HelloWorldCommand())
+            .permission("myplugin.command.helloWorld")
+            .shortDescription(Component.text("Hello World Command"))
+            .executionRequirements(context -> context.cause().context().get(EventContextKeys.PLAYER).isPresent())
+            .build();
+    }
+
+.. note::
+
+    We recommend you to add an optional ``[player]`` argument to make the command *console-friendly*. 
+    Example ``/suicide [player]``
+
+    The easiest solution for this is to append a ``playerOrTarget`` command parameter (see :doc:`argumentparsing`) 
+    instead of providing a execution requirement.
+
+.. tip::
+
+    Often times command are put in as player-only as they require the location in the world the command was
+    executed from. Best practise would be to check for the target location using :javadoc:`EventContextKeys#LOCATION`
+    instead of the player as this would allow command blocks to run the command without specifing a player.
+
+Writing a Command Executor
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The only required component to build a simple command is the command executor class, which contains the logic of 
+the command.
+
+The class has to implement the :javadoc:`CommandExecutor` interface, which contains a single method that is called on 
+command execution. The method contains a single argument of :javadoc:`CommandContext` which contains all data attached
+to the command.
+
+Example: Simple Command Executor
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: java
+
+    import org.spongepowered.api.command.CommandResult;
+    import org.spongepowered.api.command.parameter.CommandContext;
+    import org.spongepowered.api.command.CommandExecutor;
+    import org.spongepowered.api.command.exception.CommandException;
+
+    public class HelloWorldCommand implements CommandExecutor {
+    
+        @Override
+        public CommandResult execute(CommandContext context) throws CommandException{
+            context.sendMessage(Component.text("Hello World!"));
+            return CommandResult.success();
+        }
+
+    }
+
+.. tip::
+
+    You can use `anonymous classes <https://docs.oracle.com/javase/tutorial/java/javaOO/anonymousclasses.html>`_ to 
+    define the command executor in the command's build process (see example in the :doc:`argumentparsing` page).
+
+Command Result
+~~~~~~~~~~~~~~
+
+The command result is used to give more information about how the command was executed.
+In the example above we used :javadoc:`CommandResult#success` however we can give more information then this,
```

Comment:
comma needed after the javadoc reference
then -> than

Reply:
Updated